library(Gviz)
options(ucscChromosomeNames=FALSE)
alTrack <- AlignmentsTrack("SRR493366Aligned.sortedByCoord.out.bam",isPaired=T,size=0.05,background.title ="#042E8A",col="#042E8A",fill="#042E8A",fontsize=16)
grtrack <- GeneRegionTrack("/group/bioi1/shared/projects/superTranscript/reference_generation/human_super_reference.overlap_removed.gtf",chromsome="ENSG00000160072",name="Annotated Transcripts",transcriptAnnotation="transcript",size=0.4,col="black",fontsize=16)
grttrack2 <- GeneRegionTrack("human_super_reference.overlap_removed.flattened.gtf",chromsome="ENSG00000160072",name="Annotated Blocks",background.title="#00BA38",fill="#00BA38",col="black",fontsize=13)
splicetrack <- GeneRegionTrack("Spliced.gtf",chromsome="ENSG00000160072",name="Dynamic Blocks",col="black",background.title ="#619CFF",fill="#619CFF",fontsize=13)
gtrack <- GenomeAxisTrack()
pdf('sashimi.pdf',width=15,height=6)
#plotTracks(list(gtrack,alTrack,grtrack,grttrack2,splicetrack),from=-500,to=8000,chromosome="ENSG00000160072",type=c("coverage","sashimi"))
plotTracks(list(gtrack,alTrack,grtrack,grttrack2,splicetrack),from=3000,to=6500,chromosome="ENSG00000160072",type=c("coverage","sashimi"))
dev.off()
